internal/pkg/deploy/cloudformation/project_test.go

```diff
@@ -1156,6 +1156,49 @@ func TestDelegateDNSPermissions(t *testing.T) {
 				}
 			},
 		},
+
+		"Returns errChangeSetEmpty from deployChangeSet": {
+			project: &archer.Project{
+				AccountID: "1234",
+				Name:      "project",
+				Domain:    "amazon.com",
+			},
+			want: nil,
+			mockCFClient: func() *mockCloudFormation {
+				return &mockCloudFormation{
+					t: t,
+					mockCreateChangeSet: func(t *testing.T, in *cloudformation.CreateChangeSetInput) (*cloudformation.CreateChangeSetOutput, error) {
+						require.Equal(t, 6, len(in.Parameters))
+						return &cloudformation.CreateChangeSetOutput{
+							StackId: aws.String("stackname"),
+						}, nil
+					},
+					mockExecuteChangeSet: func(t *testing.T, in *cloudformation.ExecuteChangeSetInput) (*cloudformation.ExecuteChangeSetOutput, error) {
+						return nil, nil
+					},
+					mockDescribeStacks: func(t *testing.T, in *cloudformation.DescribeStacksInput) (*cloudformation.DescribeStacksOutput, error) {
+						stack := mockProjectRolesStack("stackname", map[string]string{
+							"ProjectDNSDelegatedAccounts": "1234",
+						})
+						return &cloudformation.DescribeStacksOutput{Stacks: []*cloudformation.Stack{stack}}, nil
+					},
+					mockDescribeChangeSet: func(t *testing.T, in *cloudformation.DescribeChangeSetInput) (*cloudformation.DescribeChangeSetOutput, error) {
+						return &cloudformation.DescribeChangeSetOutput{
+							ExecutionStatus: aws.String(cloudformation.ExecutionStatusAvailable),
+						}, nil
+					},
+					mockWaitUntilChangeSetCreateCompleteWithContext: func(t *testing.T, in *cloudformation.DescribeChangeSetInput) error {
+						return fmt.Errorf("error") // <- this triggers a DescribeChangeSet which will return no change set
```

Comment:
Should this be `return errChangeSetEmpty`?

Reply:
the return error is not directly bubbled up but just as a signal to indicate something is wrong and trigger a describe. the result of describe will be checked to see if `errChangeSetEmpty` should be returned.

https://github.com/aws/amazon-ecs-cli-v2/blob/4612ad5de17be87d27dced717e5206f264d1d8d8/internal/pkg/deploy/cloudformation/cloudformation.go#L286-L300